File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        res = []
        for token in tokens:
            if token in "+-*/":
                op2 = res[-1]
                res.pop()
                op1 = res[-1]
                res.pop()
                if token == "+":
                    op1 += op2
                elif token == "-":
                    op1 -= op2
                elif token == "*":
                    op1 *= op2
                elif token == "/":
                    if op2 == 0:
                        op1 = 0
                    else:
                        op1 = int(float(op1) / op2)
                res.append(op1)
            else:
                res.append(int(token))

        return res[-1]

File: Data Structures & Algorithms/climbing-stairs/submission-4.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n == 1 || n == 2)
            return n;
        int prev2 = 2;
        int prev1 = 1;
        int res;
        for(int i = 2; i < n; i++){
            res = prev2 + prev1;
            prev1 = prev2;
            prev2 = res;
        }
        return res;
    }
};
